inbm/dispatcher-agent/dispatcher/sota/snapshot.py

```diff
@@ -223,10 +227,13 @@ def recover(
         dispatcher_state.clear_dispatcher_state()
         if self.snap_num:
             self._rollback_and_delete_snap()
+            logger.debug("Rebooting to recover from failed SOTA...")
+            rebooter.reboot()
         else:
             time.sleep(time_to_wait_before_reboot)
-        logger.debug("Rebooting to recover from failed SOTA...")
-        rebooter.reboot()
+            if self._reboot_device:
+                logger.debug("Rebooting to recover from failed SOTA...")
+                rebooter.reboot()
```

Comment:
**Suggestion:** Refactor the recover method to avoid code duplication and centralize the reboot logic. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        dispatcher_state.clear_dispatcher_state()
        if self.snap_num:
            self._rollback_and_delete_snap()
        self._conditional_reboot(rebooter, time_to_wait_before_reboot)

        def _conditional_reboot(self, rebooter: Rebooter, time_to_wait_before_reboot: int) -> None:
            """Helper method to handle conditional reboot."""
            time.sleep(time_to_wait_before_reboot)
            if self._reboot_device:
                logger.debug("Rebooting to recover from failed SOTA...")
                rebooter.reboot()
```